src/parsers/manifest/smooth/create_parser.ts

```diff
@@ -581,35 +584,28 @@ function createSmoothStreamingParser(
                                         maximumTime.value),
                         time };
       }
-      const manifestDuration = root.getAttribute("Duration");
-      duration = (manifestDuration != null && +manifestDuration !== 0) ?
-        (+manifestDuration / timescale) : undefined;
-
     } else {
       periodStart = firstTimeReference != null ? firstTimeReference :
                                                  0;
       minimumTime = { isContinuous: false,
                       value: firstTimeReference != null ? firstTimeReference :
                                                           0,
                       time: performance.now() };
-
-      // if non-live and first time reference different than 0. Add first time reference
-      // to duration
-      const manifestDuration = root.getAttribute("Duration");
-
-      if (manifestDuration != null && +manifestDuration !== 0) {
-        duration = lastTimeReference == null ?
-          (+manifestDuration / timescale) +
-          (firstTimeReference === undefined ? 0 :
-                                              firstTimeReference) :
-          lastTimeReference;
-      } else {
-        duration = undefined;
+      if (lastTimeReference !== undefined) {
+        maximumTime = { isContinuous: false,
+                        value: lastTimeReference,
+                        time: performance.now() };
+      } else if (mediaPresentationTime !== undefined) {
+        maximumTime = { isContinuous: false,
+                        value: mediaPresentationTime,
+                        time: performance.now() };
```

Comment:
Nope maximumTime will be duration + minimum here (taken from the old code)

Reply:
Yes, done